src/main/java/uk/gov/hmcts/cmc/claimstore/events/claimantresponse/ClaimantResponseActionsHandler.java

```diff
@@ -71,6 +75,14 @@ private boolean isFreeMediationConfirmed(Claim claim) {
             && isOptedForMediation(response);
     }
 
+    private boolean hasClaimantSettledForFullDefense(Claim claim){
+        ClaimantResponse claimantResponse = claim.getClaimantResponse().orElseThrow(IllegalStateException::new);
+        Response response = claim.getResponse().orElseThrow(IllegalArgumentException::new);
+        return claimantResponse.getType() == ClaimantResponseType.ACCEPTATION
+            && response.getResponseType() == ResponseType.FULL_DEFENCE
```

Comment:
Is this clause needed? Both isResponseFullDefenceStatesPaid and isFullDefenceDispute check this too.